src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -173,25 +200,129 @@ public Action createAction(final Action action) {
   }
 
   @Transactional
-  public void createScheduledActions(Integer actionPlanJobId) {
-    final ActionPlanJob actionPlanJob =
-        actionPlanJobRepository.findByActionPlanJobPK(actionPlanJobId);
+  public void createScheduledActions(ActionPlan actionPlan, ActionPlanJob actionPlanJob) {
+    List<ActionCase> cases = actionCaseRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
+    List<ActionRule> rules = actionRuleRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
+
+    // For each case/rule pair create actions
+    cases.forEach(
+        caze -> {
+          if (isActionPlanLive(caze)) {
+            rules.forEach(
+                rule -> {
+                  if (hasRuleTriggered(rule)) {
+                    try {
+                      createActions(caze, rule);
+                    } catch (Exception ex) {
+                      log.with("caseId", caze.getId().toString())
+                          .with("actionRuleId", rule.getId().toString())
+                          .error(ex.toString());
```

Comment:
Do something like: `.error("Caught exception", ex);` so we get the stack trace etc.